nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java

```diff
@@ -4697,7 +4697,7 @@ public void verifyCanSaveToFlowRegistry(final String registryId, final String bu
                 // Flow ID is the same. We want to publish the Process Group as the next version of the Flow.
                 // In order to do this, we have to ensure that the Process Group is 'current'.
                 final VersionedFlowState state = vci.getStatus().getState();
-                if (state == VersionedFlowState.STALE || state == VersionedFlowState.LOCALLY_MODIFIED_AND_STALE) {
+                if (state == VersionedFlowState.STALE) {
```

Comment:
I think we need to update this conditional to allow the state of `LOCALLY_MODIFIED_AND_STALE` but only when the action is `FORCE_COMMIT`. If the action is `COMMIT`, I think we need to retain this check.

Reply:
Good catch, I pushed up some changes that should address this. Require a few additional changes to pass the action down to this point in the code, and also realized we never initially ensured the action was present and valid, so added that as well.